### PR TITLE
Improve and add more tests

### PR DESCRIPTION
**Pull request by sidmohanty11**
## Description

This PR fixes Open AI LLM for streaming and adds more tests

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -120,6 +120,7 @@ ftfy = { version = "6.1.1", optional = true }
 regex = { version = "2023.8.8", optional = true }
 huggingface_hub = { version = "^0.17.3", optional = true }
 pymilvus = "2.3.1"
+pytest-asyncio = "^0.21.1"
```
Comment (deshraj) [inline pyproject.toml:123]: It should be part of the dev dependencies.

File: tests/llm/test_huggingface.py [hunk 2]
```diff
@@ -1,64 +1,63 @@
...
+    mock_llm_instance = mocker.Mock(return_value="Test answer")
+    mocker.patch("embedchain.llm.huggingface.HuggingFaceHub", return_value=mock_llm_instance)
+
+    llm = HuggingFaceLlm(huggingface_llm_config)
+    answer = llm.get_llm_model_answer("Test query")
+
+    assert answer == "Test answer"
+    mock_llm_instance.assert_called_once_with("Test query")
+    mock_llm_instance.assert_called_once_with("Test query")
+    mock_llm_instance.assert_called_once_with("Test query")
```
Comment (deshraj) [inline tests/llm/test_huggingface.py:63]: Do we need to call this three times? I am a bit confused here 😅

File: pyproject.toml (hunk 1 above)
Comment (deshraj) [reply pyproject.toml:123]: Also, I am not sure if we really need this package btw.

Comment (sidmohanty11) [reply pyproject.toml]: @deshraj thank you, I've added it to dev dependency. It's actually needed by a test for the Poe bot, which uses async code that's not currently running and is being skipped.

File: tests/llm/test_huggingface.py (hunk 2 above)
Comment (sidmohanty11) [reply tests/llm/test_huggingface.py]: Sorry for the miss, yup you're right! I've removed it :)

**Review by deshraj — APPROVED** [review]

**Pull request merged**